.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,51 @@
+name: facebook/metro/build-and-test
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+# head_ref is per PR, so this ensures that updating the latest PR commit
+# will cancel the previous run of the workflow and trigger a new one
+concurrency:
+  group: "build-and-test-${{ github.head_ref }}"
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  run-js-checks:
+    runs-on: ubuntu-latest
+    name: "Typescript, Lint, Smoke Test"
```

Comment:
nit: This is _Flow_, TypeScript, Lint ... — can we genericise?
```suggestion
    name: "Static analysis"
```

or "Lint", "Run JS checks"

Reply:
This used to be `Type check, lint, smoke test`, which is probably more accurate tbh. But yeah, it might be cleaner if we separated the smoke test from static analysis.